plaso/output/elastic_ts.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+"""An output module that saves events to Elasticsearch TS index."""
+
+from plaso.output import logger
+from plaso.output import manager
+from plaso.output import shared_elastic
+
+
+class ElasticTimesketchOutputModule(
+    shared_elastic.SharedElasticsearchOutputModule):
+  """Output module for Timesketch Elasticsearch."""
+
+  NAME = 'elastic_ts'
+  DESCRIPTION = (
+      'Saves the events into an Elasticsearch database for use '
+      'with Timesketch.')
+
+  MAPPINGS_FILENAME = 'timesketch.mappings'
+
+  def __init__(self, output_mediator):
+    """Initializes a Timesketch output module.
+
+    Args:
+      output_mediator (OutputMediator): mediates interactions between output
+          modules and other components, such as storage and dfvfs.
+    """
+    super().__init__(output_mediator)
+    self._timeline_id = None
+
+  def _GetSanitizedEventValues(
+      self, event, event_data, event_data_stream, event_tag):
+    """Sanitizes the event for use in Elasticsearch.
+
+    The event values need to be sanitized to prevent certain values from
+    causing problems when indexing with Elasticsearch. For example the path
+    specification is a nested dictionary which will cause problems for
+    Elasticsearch automatic indexing.
+
+    Args:
+      event (EventObject): event.
+      event_data (EventData): event data.
+      event_data_stream (EventDataStream): event data stream.
+      event_tag (EventTag): event tag.
+
+    Returns:
+      dict[str, object]: sanitized event values.
+
+    Raises:
+      NoFormatterFound: if no event formatter can be found to match the data
+          type in the event data.
+    """
+    event_values = super()._GetSanitizedEventValues(
+        event=event, event_data=event_data,
+        event_data_stream=event_data_stream, event_tag=event_tag)
+
+    if self._timeline_id:
+      event_values['__ts_timeline_id'] = self._timeline_id
+
+    return event_values
+
+  def GetMissingArguments(self):
+    """Retrieves a list of arguments that are missing from the input.
+
+    Returns:
+      list[str]: names of arguments that are required by the module and have
+          not been specified.
+    """
+    if not self._timeline_id:
+      return ['timeline_id']
+    return []
+
+  def SetTimelineID(self, timeline_id):
```

Comment:
we prefer not to use abbreviations, please change ID into Identifier

Reply:
done